public/Test-AdfLinkedService.ps1

```diff
@@ -9,27 +9,37 @@ function Test-AdfLinkedService {
         [String] $ResourceGroupName,
         [parameter(Mandatory = $true)] 
         [String] $SubscriptionID,
-        [parameter(Mandatory = $true)] 
+        [parameter(Mandatory = $true, ParameterSetName="ClientDetails")] 
         [String] $TenantID,
-        [parameter(Mandatory = $true)] 
+        [parameter(Mandatory = $true, ParameterSetName="ClientDetails")] 
         [String] $ClientID,
-        [parameter(Mandatory = $true)] 
-        [String] $ClientSecret
+        [parameter(Mandatory = $true, ParameterSetName="ClientDetails")] 
+        [String] $ClientSecret,
+        [parameter(Mandatory = $true, ParameterSetName="AzRestMethod")]
+        [switch] $preferAzRestMethod
```

Comment:
I can't see the role of this switch

Reply:
Another option would be to use Az context of service principal details are not present in the command-line.

See other comments for more details.